oap-server/server-library/library-server/src/main/java/org/apache/skywalking/oap/server/library/server/http/HTTPServerConfig.java

```diff
@@ -39,4 +39,14 @@ public class HTTPServerConfig {
     private int acceptQueueSize = 0;
     @Builder.Default
     private int maxRequestHeaderSize = 8192;
+
+    @Builder.Default
+    private boolean enableTLS = false;
+    @Builder.Default
+    private boolean enableTlsSelfSigned = false;
+
+    private int httpsPort;
+    private String tlsKeyPath;
+    private String tlsCertChainPath;
```

Comment:
All these configurations are domain-name:port specific, i.e. every domain name:port should have the corresponding https port, tlsKeyPath, tlsCertChainPath.

Reply:
One question, today's gRPC seems not setting the domain name?

Reply:
> One question, today's gRPC seems not setting the domain name?

We have no built-in https support in today's gRPC server inside OAP. In another word, users have to setup a gateway to support https for gRPC port 11800.

Reply:
<img width="808" alt="image" src="https://user-images.githubusercontent.com/5441976/213448204-3d859dc6-690c-4532-b51c-27f3732c9bc6.png">

I mean we have relative/similar configurations in gRPC part.

Reply:
We are actually using the server host here 

https://github.com/apache/skywalking/blob/289427990aa857298e498782ad97a1d8db68ae4a/oap-server/server-library/library-server/src/main/java/org/apache/skywalking/oap/server/library/server/grpc/GRPCServer.java#L86

After thinking twice I think I was wrong, we don't need to configure the host/domain name, as long as the  CN field in the cert is the same as the hostname that are used to access this endpoint, we are good to go.

Reply:
But still, we need to make it clear, do we support https for all other http server like Prometheus, Zipkin receiver, or do we want to only support that for AWS receiver? If we want to support for all, we need these configurations for each server respectively, otherwise we should move these configurations to that plugin.

Reply:
> But still, we need to make it clear, do we support https for all other http server like Prometheus, Zipkin receiver, or do we want to only support that for AWS receiver? If we want to support for all, we need these configurations for each server respectively, otherwise we should move these configurations to that plugin.

For now, we plan to build a new HTTPS receiver inside `aws-firehose-receiver` separate from the existing core and sharing server HTTP server, also separate from Prometheus or Zipkin server.
We don't need to support all in HTTPS, at least not for now.

Reply:
So I don't need to change configuration item now ?

Reply:
We don't need to add a domain name for now. But `enableTlsSelfSigned` seems to need to be removed.